nestkernel/conn_builder.cpp

```diff
@@ -369,13 +371,29 @@ nest::ConnBuilder::change_connected_synaptic_elements( index sgid,
 void
 nest::ConnBuilder::connect()
 {
+  if ( symmetric_ && not supports_symmetric() )
+    throw NotImplemented(
+      "So far Connect doesn't support symmetric connections for this "
+      "connection rule." );
+
   if ( pre_synaptic_element_name != "" && post_synaptic_element_name != "" )
   {
+    if ( symmetric_ )
+      throw NotImplemented(
+        "So far Connect doesn't support symmetric connections in combination "
+        "with structural plasticity." );
     sp_connect_();
   }
   else
   {
     connect_();
+    if ( symmetric_ )
+    {
+      std::swap( sources_, targets_ );
+      connect_();
+      std::swap( sources_,
+        targets_ ); // not really necessary, but leaves things in clean state
```

Comment:
Move the comment to a separate line, so that the code fits on one line for clarity. The "not really necessary" is a bit misleading. This is only the case if you call `connect()` only a single time on a `ConnBuilder` object. In case of multiple calls (which are not expressly forbidden anywhere), swapping back is essential.